### DISCUSSION TITLE
Can I use filter Jinja Templating for schema selection in a database?

### DISCUSSION THREAD

**Discussion by vamshi975** [apache/superset#18339] [Q&A / Help]
I have a scenario where we have 10,000 schemas in one database and I am trying to create a dynamic dashboard with plots that are common throughout all schemas. I am looking for an effective solution by selecting the schema name as a filter(I did creating a filter with all schema names in it) and configure charts using jinja templatíng for schema selection in SQL query.  

Also, is it possible to use plpgsql in sql editor when using postgresql as a datasource? 

I am new to superset and looking for any suggestions?

Thanks in advance.

**Comment by mistercrunch** [comment]
This seems worthy at a high level, so here's pointer to `get_template_processor` and how it's used to compile expressions in columns, filters and other places. Conceptually it seems like it could make sense to do this on the database schema too, but has security / data access policy implications, and it could be slightly complicated from an RBAC standpoint as I'm assuming the template (NOT rendered) will be shown as a single resource. IDK more research would be needed.
 
https://github.com/apache/superset/blob/5aaa3330a66c90047da3dcc557970cf9383d4933/superset/connectors/sqla/models.py#L84

**Comment by vamshi975** [comment]
@mistercrunch Thank you for the quick reply. 

Also, I am interested to know if pl/pgsql statements are supported in superset ?? I couldn't find anything related to that in superset documentation.

**Comment by villebro** [comment]
@bvy007 plsql should work in SQL Lab, but probably doesn't work properly in virtual tables (it's been a while since I've tested Superset on an Oracle database).
